protocols/grpc/utils.go

```diff
@@ -287,3 +299,86 @@ func ResourcesGRPCtoOCI(grpcResources *LinuxResources) (*specs.LinuxResources, e
 
 	return s, err
 }
+
+// ChangeToBundlePath changes the cwd to the bundle path defined in the OCI spec
+func ChangeToBundlePath(spec *specs.Spec) (string, error) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return cwd, err
+	}
+
+	if spec == nil || spec.Root == nil || spec.Root.Path == "" {
+		return cwd, errors.New("Invalid OCI spec")
+	}
+
+	rootfsPath := spec.Root.Path
+	if !filepath.IsAbs(rootfsPath) {
+		rootfsPath = filepath.Join(cwd, rootfsPath)
+	}
+
+	bundlePath := filepath.Dir(rootfsPath)
+
+	err = os.Chdir(bundlePath)
+	return cwd, err
+}
+
+// WriteSpecToFile writes the container's OCI spec to 'config.json'
+// in the bundle path as expected by the OCI specification.
+func WriteSpecToFile(spec *specs.Spec) error {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+
+	configPath := path.Join(cwd, ociConfigFile)
+	f, err := os.OpenFile(configPath, os.O_RDWR|os.O_CREATE, ociConfigFileMode)
+	if err != nil {
+		return err
+	}
+
+	defer f.Close()
+	err = json.NewEncoder(f).Encode(&spec)
+	return err
+}
+
+func isValidHook(file os.FileInfo) bool {
+	if file.IsDir() {
+		return false
+	}
+
+	mode := file.Mode()
+	if (mode & os.ModeSymlink) != 0 {
+		return false
+	}
+
+	// check that this file is executable
+	perm := mode & os.ModePerm
+	return (perm & 0111) != 0
+}
+
+// FindHooks searches guestHookPath for any OCI hooks for a given hookType
+func FindHooks(guestHookPath, hookType string) (hooksFound []specs.Hook) {
+	hooksPath := path.Join(guestHookPath, hookType)
```

Comment:
So `prestart` hook will be in `guestHookPath/prestart` and `poststart` hook will be in `guestHookPath/poststart`, 
and `guestHookPath` is specified by kata-runtime cli configuration, right?

I think we need some documents to teach user, we can't expect user can find this hidden advanced feature.

Reply:
+1 ;)